File: 프로그래머스/0/120834. 외계행성의 나이/외계행성의 나이.py
def solution(age):
    # 1. 변환된 결과를 저장할 변수 초기화
    answer = ''

    # 2. 나이를 문자열로 변환 후 각 자리 숫자를 리스트로 저장
    my_age = list(map(int, str(age)))

    # 3. 각 숫자를 대응되는 알파벳으로 변환
    for m in my_age:
        answer += chr(m + ord('a'))  # 'a'의 ASCII 코드(97) + 숫자

    # 4. 변환된 문자열 반환
    return answer

# ✅ 예제 테스트 실행
print(solution(23))   # 결과: "cd"
print(solution(51))   # 결과: "fb"
print(solution(100))  # 결과: "baa"

File: 프로그래머스/0/120834. 외계행성의 나이/외계행성의 나이.java
public class Solution {

    public static String solution(int age) {
        StringBuilder answer = new StringBuilder();

        // 1. 나이를 문자열로 변환하여 각 자리 문자 순회
        String ageStr = String.valueOf(age);

        for (int i = 0; i < ageStr.length(); i++) {
            char digit = ageStr.charAt(i);
            // 2. '0'을 기준으로 문자 → 정수 → 알파벳 문자로 변환
            int num = digit - '0'; // 문자 '3' → 정수 3
            char ch = (char) (num + 'a'); // 0 → 'a', 1 → 'b', ...
            answer.append(ch);
        }

        return answer.toString();
    }

    // ✅ 테스트 실행
    public static void main(String[] args) {
        System.out.println(solution(23));   // cd
        System.out.println(solution(51));   // fb
        System.out.println(solution(100));  // baa
    }
}
